server/container/core/src/main/java/org/apache/james/server/core/MailImpl.java

```diff
@@ -398,11 +398,11 @@ private static void detectPossibleLoop(String currentName, int loopThreshold, ch
      */
     private PerRecipientHeaders perRecipientSpecificHeaders;
 
-    private MailImpl(String name,
+    private MailImpl(String state,
                      Map<AttributeName, Attribute> attributes,
                      List<MailAddress> recipients,
                      PerRecipientHeaders perRecipientHeaders) {
-        setName(name);
+        setState(state);
```

Comment:
ah, that's it :) so where is set the mail name with this constructor?

Reply:
I didn't included the requirement of the name in this PR because I'd like it to build and be merged in a short delay.
In another one, I'll enforce the requirement of having a well-defined name (I fear that having a fallback will lead to more bugs so I don't want to go that way).